dns_sd_windows.c

```diff
@@ -47,6 +47,74 @@ static bool is_localhost4(const struct sockaddr_in *saddr)
 		saddr->sin_addr.S_un.S_un_b.s_b4 == 1;
 }
 
+static struct dns_sd_discovery_data *new_discovery_data(struct dns_sd_discovery_data *dd)
+{
+	struct dns_sd_discovery_data *d;
+
+	d = zalloc(sizeof(*d));
+	if (!d)
+		return NULL;
+
+	if (dd)
+		d->lock = dd->lock;
+
+	return d;
+}
+
+static mdns_string_t ipv4_address_to_string(char* buffer, size_t capacity, const struct sockaddr_in* addr,
+	size_t addrlen) {
+	char host[NI_MAXHOST] = { 0 };
+	char service[NI_MAXSERV] = { 0 };
+	int ret = getnameinfo((const struct sockaddr*)addr, (socklen_t)addrlen, host, NI_MAXHOST,
+		service, NI_MAXSERV, NI_NUMERICSERV | NI_NUMERICHOST);
```

Comment:
Avoid making non-trivial initializations in the variable definitions - just declare `int ret;` here and initialize it below.

Reply:
changed this around / so resolved.